### DISCUSSION TITLE
Having to reload page for session to populate after login?

### DISCUSSION THREAD

**Discussion by kWAYTV** [nextauthjs/next-auth#11801] [Help]
The site is [open source](https://github.com/habbofun/hfun-battleball-leaderboard), any fix would be greatly appreciated, i'll keep updated if i fix the issue, please @ me for anything.

Repo: https://github.com/habbofun/hfun-battleball-leaderboard

Important files:
https://github.com/habbofun/hfun-battleball-leaderboard/blob/master/src/hooks/use-current-user.ts
https://github.com/habbofun/hfun-battleball-leaderboard/blob/master/src/server/auth.config.ts
https://github.com/habbofun/hfun-battleball-leaderboard/blob/master/src/app/(protected)/settings/page.tsx
https://github.com/habbofun/hfun-battleball-leaderboard/blob/master/src/components/auth/login/login-form.tsx

Also found out it's not just between login/logout, if i update my database from prisma studio, i have to re-login to see the changes.

**Comment by jaxomlotus** [comment]
This is a massive security hole unfortunately. When signing out, there should no longer be a token with PII installed inside it in a cookie. this means that anyone accessing a nextauth secured site from a public computer cannot rely on signout to remove their information.

**Reply by kWAYTV** [reply]
just moved to clerk, problem solved.
